app/policies/application_policy.rb

```diff
@@ -59,5 +59,34 @@ def initialize(user, scope)
     def resolve
       scope.all
     end
+
+    def graphql_id_query
+      id = user.graphql_params.id
+
+      item = scope.find(id)
+      raise Exceptions::NotAuthorizedError, "Read access not authorized for request #{item.id}" unless resource_check('read', item)
+
+      scope.where(:id => id)
+    end
+
+    def graphql_filter_query
+      scope.where(user.graphql_params.filter)
+    end
+
+    def graphql_collection_query
```

Comment:
you only need
```
  def graphql_collection_query
    scope.where(user.graphql_params&.filter)
  end
```

Methods `graphql_query_by_filter?` and `graphql_filter_query` can go away.

Reply:
👍